ros/src/tl_detector/tl_detector.py

```diff
@@ -145,14 +145,13 @@ def process_traffic_lights(self):
         closest_light = None
         if self.pose:
             car_pos = self.get_closest_waypoint(self.pose.pose)
-
-        stop_wp = self.get_stop_waypoint(car_pos)
-        if self.is_tl_visible(stop_wp, car_pos):
-            thread = Thread(
-                    target = self.predict_and_publish, 
-                    args=[stop_wp],
-                    )
-            thread.start()
+            stop_wp = self.get_stop_waypoint(car_pos)
+            if self.is_tl_visible(stop_wp, car_pos):
+                thread = Thread(
```

Comment:
Can we maybe have the thread in a different node ?
I think it it will be safer.

Reply:
Yes, that would be a good solution.